### PR TITLE
api: Fix issue observed with message publish on creation of domain

### PR DESCRIPTION
**Pull request by Pearl1594**
### Description

This PR fixes an error observed during creation of a domain
```
2022-03-15 09:30:04,134 INFO  [c.c.u.DomainManagerImpl] (qtp1007412025-20:ctx-a68b2766 ctx-acce7b84) (logid:c9fdb68b) Domain UUID [5180d5f9-5b60-4034-92e4-0131db9c0e70] generated for domain name [test].
2022-03-15 09:30:04,156 ERROR [o.a.c.f.m.MessageBusBase] (qtp1007412025-20:ctx-a68b2766 ctx-acce7b84) (logid:c9fdb68b) NO EVENT PUBLISH CAN BE WRAPPED WITHIN DB TRANSACTION!
com.cloud.utils.exception.CloudRuntimeException: NO EVENT PUBLISH CAN BE WRAPPED WITHIN DB TRANSACTION!
	at org.apache.cloudstack.framework.messagebus.MessageBusBase.publish(MessageBusBase.java:167)
	at com.cloud.user.DomainManagerImpl$1.doInTransaction(DomainManagerImpl.java:223)
	at com.cloud.user.DomainManagerImpl$1.doInTransaction(DomainManagerImpl.java:216)
	at com.cloud.utils.db.Transaction$2.doInTransaction(Transaction.java:50)
	at com.cloud.utils.db.Transaction.execute(Transaction.java:40)
	at com.cloud.utils.db.Transaction.execute(Transaction.java:47)
	at com.cloud.user.DomainManagerImpl.createDomain(DomainManagerImpl.java:216)
	at com.cloud.user.DomainManagerImpl.createDomain(DomainManagerImpl.java:205)

```

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

**Review by weizhouapache — APPROVED** [review]

**Review by rohityadavcloud — APPROVED** [review]

File: server/src/main/java/com/cloud/user/DomainManagerImpl.java [hunk 1]
```diff
@@ -220,11 +220,10 @@ public DomainVO doInTransaction(TransactionStatus status) {
                 _resourceCountDao.createResourceCounts(domain.getId(), ResourceLimit.ResourceOwnerType.Domain);
 
                 CallContext.current().putContextParameter(Domain.class, domain.getUuid());
-                _messageBus.publish(_name, MESSAGE_ADD_DOMAIN_EVENT, PublishScope.LOCAL, domain.getId());
                 return domain;
             }
         });
-
+        _messageBus.publish(_name, MESSAGE_ADD_DOMAIN_EVENT, PublishScope.LOCAL, domain.getId());
```
Comment (rohityadavcloud) [inline server/src/main/java/com/cloud/user/DomainManagerImpl.java:226]: Just wondering if this is cause for concern, what if the transaction fails for some reason @Pearl1594, should we check for that (for ex. the domain ID is not null) ?

Comment (Pearl1594) [reply server/src/main/java/com/cloud/user/DomainManagerImpl.java]: Thanks @rohityadavcloud . I've added a check

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Pull request merged**